inbm/genai_unit_tests/downloader_genai_unit_test.py

```diff
@@ -0,0 +1,94 @@
+import sys
+sys.path.append('/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm/')
+sys.path.append('/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm/dispatcher-agent/')
+sys.path.append('/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm-lib/')
+sys.path.append('/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm/dispatcher-agent/dispatcher/')
+import pytest
+from unittest.mock import MagicMock, patch, mock_open
+from dispatcher.downloader import _check_if_valid_file
+from dispatcher.dispatcher_exception import DispatcherException
+from unittest.mock import Mock, patch
+from dispatcher.downloader import download
+from dispatcher.packagemanager.local_repo import DirectoryRepo
+from inbm_common_lib.utility import CanonicalUri
+
+#DO NOT DELETE THIS LINE - TestCheckIfValidFile
+'''
+ADD HUMAN FEEDBACK BELOW:
+
+'''
+class TestCheckIfValidFile:
```

Comment:
To go back to Pull Request Summary table, Click on the [link](https://github.com/ankit-r-gajera/genai_ag_intel-inb-manageability/pull/8#)

<h4>For feedback purposes, please include both emoticon-based and qualitative feedback in your reply to this comment.</h4>

To go back to Pull Request Summary table, Click on the [link](https://github.com/ankit-r-gajera/genai_ag_intel-inb-manageability/pull/8#)